web/app/tournament/tourny.py

```diff
@@ -169,3 +172,196 @@ def select_player(tournament_id, tournament_player_id):
         # flash("Could not delete player: {{ tp.name }}")
         return redirect(url_for('.players_index',
             tournament_id=tournament_id))
+
+
+@tournament.route('/<tournament_id>/matches/', methods=['GET'])
+def matches_index(tournament_id):
+    any_matches = Match.query.count()
+    matches = Match.query.filter_by(
+        tournament_id=tournament_id).order_by("t_round, id")
+    if any_matches > 0:
+        return render_template('tournament_match_index.html',
+        tournament_id=tournament_id,
+        matches=matches)
+    else:
+        #  flash('URL Error')
+        return redirect(url_for('.show', tournament_id=tournament_id))
+
+
+@tournament.route('/<tournament_id>/round/<round_id>/', methods=['GET'])
+@tournament.route('/<tournament_id>/round/<round_id>/matches/', methods=['GET'])
+def matches_by_round_index(tournament_id, round_id):
+    ###
+    # this ugly piece allows for conditional in the template
+    # if viewed round is latest_round
+    latest_round = False
+    round_in_url = round_id
+    if round_in_url == str(Round.query.all()[-1].id):
+        latest_round = True
+        ###
+    any_matches = Match.query.filter_by(tournament_id=tournament_id,
+                                        t_round=round_id).count()
+    matches = Match.query.filter_by(tournament_id=tournament_id,
+                                    t_round=round_id)
+    if any_matches > 0:
+        return render_template('tournament_matches_by_round.html',
+        tournament_id=tournament_id,
+        matches=matches,
+        round_id=round_id,
+        latest_round=latest_round)
+    else:
+        #  flash('URL Error')
+        return redirect(url_for('.show', tournament_id=tournament_id))
+
+
+@tournament.route('/<tournament_id>/full_pairing/', methods=['POST'])
+def full_pairing(tournament_id):
+    """Currently the full_pairing method takes all tournament players,
+    shuffles them, and pairs them each into a match object"""
+    tournament = Tournament.query.get(tournament_id)
+    players = TournamentPlayer.query.filter_by(tournament_id=tournament_id)
+    players_list = list(players)
+    shuffle(players_list)
+
+    tourny_round = Round(tournament_id=tournament_id)
+    db.session.add(tourny_round)
+    db.session.commit()
+    tourny_round = Round.query.all()[-1]
+    while len(players_list) > 0:
+        if len(players_list) % 2 == 0:
```

Comment:
I would pop the players off right at the top of this method. As it stands, it would be really easy to miss a 'pop' and loop forever